storagemarket/deal_execution.go

```diff
@@ -66,20 +82,30 @@ func (p *Provider) doDeal(deal *types.ProviderDealState, dh *dealHandler) {
 	// will fail, we can look into it when we implement deal completion.
 }
 
-func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter, deal *types.ProviderDealState, dh *dealHandler) error {
+func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter, deal *types.ProviderDealState, dh *dealHandler) *dealMakingError {
 	// publish "new deal" event
+	// TODO Do we need to fire this event for resumed deals ?
 	p.fireEventDealNew(deal)
-
 	// publish an event with the current state of the deal
 	p.fireEventDealUpdate(pub, deal)
 
-	p.addDealLog(deal.DealUuid, "Deal Accepted")
+	p.addDealLog(deal.DealUuid, "Deal Execution Started")
 
 	// Transfer Data
 	if deal.Checkpoint < dealcheckpoints.Transferred {
 		if err := p.transferAndVerify(dh.transferCtx, pub, deal); err != nil {
 			dh.transferCancelled(nil)
-			return fmt.Errorf("execDeal failed data transfer: %w", err)
+			// if the transfer failed because of context cancellation and the context was not
+			// cancelled because of the user explicitly cancelling the transfer, this is a recoverable error.
+			if xerrors.Is(err, context.Canceled) && !dh.TransferCancelledByUser() {
+				return &dealMakingError{
+					recoverable: true,
+					err:         fmt.Errorf("execDeal failed data transfer: %w", err),
```

Comment:
This message is going to show up in the UI so I'd suggest we give as much detail as possible to help them figure out why the transfer failed:
```suggestion
					err:         fmt.Errorf("data transfer to client %s (peer ID %s) failed after %d bytes with error: %w", err),
```
(It's probably not a bad idea to include all this info in the logs as well)

Reply:
done.